crates/kubelet-derive/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+//! A crate for deriving state machine traits in Kubelet. Right now this crate only consists of a
+//! derive macro for the `TransitionTo` trait. In addition to the `derive` attribute, this macro
+//! also requires the use of a custom attribute called `transition_to` that specifies the types that
+//! can be transitioned to. Not specifying this attribute will result in a compile time error. A
+//! simple example of this is below:
+//!
+//! ```rust,no_run
+//! use kubelet_derive::TransitionTo;
+//!
+//! pub struct VolumeMount;
+//! pub struct ImagePullBackoff;
+//!
+//! #[derive(Default, Debug, TransitionTo)]
+//! #[transition_to(VolumeMount, ImagePullBackoff)]
+//! pub struct ImagePull;
+//!```
+
+extern crate proc_macro;
+
+use crate::proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput, Error, Meta};
+
+const ATTRIBUTE_NAME: &str = "transition_to";
+
+#[proc_macro_derive(TransitionTo, attributes(transition_to))]
+pub fn derive_transition_to(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+
+    let name = &input.ident;
+    let generics = &input.generics;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+    let mut token_stream = TokenStream::new();
+
+    // We need to check if we found at least one attribute. If it isn't set, we need to error
+    let mut found_attr = false;
+    for attr in input.attrs.into_iter() {
+        if let Meta::List(parsed_attr) = attr.parse_meta().unwrap() {
```

Comment:
Is the `unwrap` idiomatic?  It won't cause a panic on a malformed attribute (e.g. while typing and the RLS runs the compiler in the background)?

Reply:
Based on examples I found it is. But as I break this up, I think I can use a compile error instead